packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -968,6 +995,96 @@ export class SnapController extends BaseController<
     );
   }
 
+  #handlePreinstalledSnaps(preinstalledSnaps: PreinstalledSnap[]) {
+    for (const { snapId, manifest, files, removable } of preinstalledSnaps) {
+      const existingSnap = this.get(snapId);
+      const isAlreadyInstalled = existingSnap !== undefined;
+      const isUpdate =
+        isAlreadyInstalled && gtVersion(manifest.version, existingSnap.version);
+
+      // Disallow downgrades and overwriting non preinstalled snaps
```

Comment:
If the preinstalled Snap version is newer than the existing non-preinstalled Snap, do we still want to disallow it?

Reply:
I think so, I want to avoid a situation where we overwrite a user's manually installed snaps at all cost